compiler/src/dotty/tools/dotc/transform/init/Semantic.scala

```diff
@@ -495,6 +496,29 @@ class Semantic {
           Result(value2, errors)
       }
     }
+
+    def accessLocal(sym: Symbol, default: Result, klass: ClassSymbol, source: Tree): Contextual[Result] =
```

Comment:
It's better to remove `default` here, as it simplifies the protocol.

Reply:
Do you mean to just hardcode it as Hot?

Oh, I see: you mean to move the whole `case tmref ...` from `cases` to `accessLocal`. That case is what defines `default` to be `Hot`.